core/src/main/java/hudson/model/Computer.java

```diff
@@ -705,11 +754,12 @@ public void setTemporarilyOffline(boolean temporarilyOffline) {
      *      offline.
      */
     public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause) {
-        offlineCause = temporarilyOffline ? cause : null;
+        temporarilyOfflineCause = temporarilyOffline ? cause : null;
+        offlineCause = temporarilyOfflineCause;
```

Comment:
Can you help me understand the reason for this? I am not necessarily saying that it is wrong, but I am struggling to understand why this is necessary, especially given the unclear Javadocs surrounding the semantics of these fields. At first glance it seems confusing to me that we would go through all this trouble to separate these fields only to set them to the same value. Is this just so that `getOfflineCause()` continues to compatibly maintain its existing contract to return either the temporarily offline cause or the offline cause? If so, would it not be clearer to avoid conflating the semantics of the two fields by instead adding conditional logic in the `getOfflineCause()` getter to return the appropriate field? I apologize in advance if I am misunderstanding something, as it took me quite a while to read and understand this messy code.

Reply:
Yes I set it so that is compatible to the existing behaviour. Moving that logic to getOfflineCause should be no problem.
There are plugins that put agents to offline mode by calling setTemporarilyOffline and later they use getOfflineCause or getOfflineCauseReason to check if the offline cause is the same that they used before (ec2-plugin, vsphere-cloud-plugin, jclouds-plugin).
In core this was also done e.g. in AbstractNodeMonitorDescriptor, which could have lead to the effect that the monitor puts the agent offline, then the connection is lost and the offlinecause is overwritten, the connection comes back and the nodemonitor detects that the problem is done. But it would not take it back online because the cause is no longer what the monitor set. This change would fix this.

Reply:
> If so, would it not be clearer to avoid conflating the semantics of the two fields by instead adding conditional logic in the `getOfflineCause()` getter to return the appropriate field?
My change to do it like this now fails the tests. This is because the tests do not start the agent which causes an offlineCause to be set it seems. And thus differs from the temporarilyOfflineCause that was set.
Technically this behaviour is correct I would say, but we don't know if this could have a side effect on the plugins that read the offline cause.
So reverting this for now